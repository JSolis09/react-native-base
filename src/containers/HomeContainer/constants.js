export const DEFAULT_ACTION = 'HomeContainer/DEFAULT_ACTION';