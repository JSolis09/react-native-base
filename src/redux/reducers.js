import { combineReducers } from "redux";
import HomeContainerReducer from "../containers/HomeContainer/reducer";
import LoginContainerReducer from "../containers/LoginContainer/reducer";

export default combineReducers({
  home: HomeContainerReducer,
  auth: LoginContainerReducer
});
