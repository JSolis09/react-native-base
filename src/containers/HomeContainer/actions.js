import { DEFAULT_ACTION, MESSAGE_CHANGED } from './constants';
export const defaultAction = () => ({
    type: DEFAULT_ACTION
});

export const changeMessageAction = (message) => ({
    type: MESSAGE_CHANGED,
    message
});