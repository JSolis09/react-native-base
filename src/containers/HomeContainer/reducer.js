import { DEFAULT_ACTION, MESSAGE_CHANGED } from './constants';

const initialState = {
    message: 'Default Message'
};

export default function HomeContainerReducer(state = initialState, action) {
    switch(action.type) {
        case DEFAULT_ACTION:
            return state;
        case MESSAGE_CHANGED:
            return Object.assign({}, state, {message: action.message});
        default:
            return state;
    }
}