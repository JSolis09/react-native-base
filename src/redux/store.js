import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from 'redux-saga';
import Reducer from "./reducers";

import homeSagas from '../containers/HomeContainer/sagas';
import loginSagas from '../containers/LoginContainer/sagas';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(Reducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(homeSagas);
sagaMiddleware.run(loginSagas);

export default store;