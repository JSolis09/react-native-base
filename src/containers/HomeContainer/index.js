import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";

import Home from '../../components/Home';
import { defaultAction } from './actions'

class HomeContainer extends Component {
    static propTypes = {
        changeMessage: PropTypes.func.isRequired
    };
    static navigationOptions = {
        header: null,
    };

    componentDidMount() {
        this.props.changeMessage();
    }

    render() {
        return(<Home {...this.props} />);
    }
}

const mapStateToProps = state => ({
    message: state.home.message
});
const mapDispatchToProps = dispatch => ({
    changeMessage: () => dispatch(defaultAction())
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer);