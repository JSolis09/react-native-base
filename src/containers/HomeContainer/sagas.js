import { all, put, takeEvery } from 'redux-saga/effects';
import { DEFAULT_ACTION } from './constants';
import { changeMessageAction } from './actions';

function delay(ms) {
    return new Promise((res) => setTimeout(res, ms));
}

export function* defaultSaga() {
    const message = 'Message Changed with Saga!!';
    yield delay(1500);
    yield put(changeMessageAction(message));
}

export default function* rootSaga() {
    yield all([
        takeEvery(DEFAULT_ACTION, defaultSaga)
    ]);
}