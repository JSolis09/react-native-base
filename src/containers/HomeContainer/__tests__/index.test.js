import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from "redux";

import Reducer from '../../../redux/reducers';
import HomeContainer from '../index';

jest.mock('../../../components/Home', () => require('../../../components/Home/__mocks__/Home') );

describe('HomeContainer', () => {
    let store = createStore(Reducer);
    it('should render correctly', () => {
        const tree = create(
            <Provider store={store}>
                <HomeContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});