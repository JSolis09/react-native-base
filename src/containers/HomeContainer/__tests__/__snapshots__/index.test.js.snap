// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should render correctly 1`] = `<View />`;
